### DISCUSSION TITLE
Run a Fully Off-Chain Optimism Network with Manual Batch Submission to Ethereum Mainnet

### DISCUSSION THREAD

**Discussion by doukhahmed** [ethereum-optimism/developers#571] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hello Optimism community! I'm exploring the possibility of developing a fully off-chain Optimism network where nodes operate entirely off-chain, with batch transactions being manually submitted to the Ethereum mainnet. Is it feasible to configure and run such an off-chain Optimism network, and if so, what would be the key technical challenges and considerations? Any guidance or insights would be greatly appreciated!

**Comment by imtipi** [comment]
can you explain more? op stack is a chain,and I can't understand whats the usage to remove the chain function

> nodes operate entirely off-chain

do you mean the sequencer node?node itself already run in server(so I don't think this means on chain)

**Reply by doukhahmed** [reply]
Thank you for your response! 
What I’m aiming to do is to leverage a local network deployed in VMs to create a local Optimism network with OP and Ethereum nodes. The idea is to execute and validate transactions locally on this network, and then manually anchor these transactions to the Ethereum mainnet for final validation, thereby benefiting from Layer 1 security. I understand that OP Stack is designed to work as a chain, but my goal is to experiment with a setup where the local network operates independently, and only batches of transactions are submitted to the mainnet.
 Does this approach make sense, and are there any existing tools or configurations within the OP Stack that could support this kind of setup?

**Reply by imtipi** [reply]
> The idea is to execute and validate transactions locally on this network

currently all transactions is process by sequencer(which is permissioned,control by chain owner),you already executing locally if you run sequencer yourself on your own op stack chain

> only batches of transactions are submitted to the mainnet

op stack is
you can learn more here https://docs.optimism.io/stack/protocol/rollup/overview#block-production

**Reply by doukhahmed** [reply]
Is it possible to validate transactions off-chain in my local Optimism network with zero transaction fees, meaning I wouldn’t need a real amount of Ether for these operations? I want to ensure that my local nodes can process and validate transactions without requiring any actual ETH, as the final batch submission to the Ethereum mainnet would be done manually. Is this feasible?

**Reply by imtipi** [reply]
can set the network to zero gas,cc @opfocus do you know is that work?

> as the final batch submission to the Ethereum mainnet would be done manually

however, batches need to be published to the Ethereum mainnet regularly, and cannot be manually published at unpredictable intervals

**Reply by opfocus** [reply]
This sounds a bit like a private network that doesn't require a bridge to or from L1
Submitting a batch means spending eth on L1 (alt-da may have ideas to save this cost).
0gas can be set on L2.
I feel like manual vs. automatic doesn't seem that important.
I personally think it's possible, but I'm not sure.   CC @joohhnnn

**Reply by joohhnnn** [reply]
It seems like you're working on something quite unique. First off, the data you send to L1 directly impacts your costs related to L2 data. This means that even if you set the gas price to zero, you'll still need to cover the transaction fees for batching these transactions.
